clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -160,17 +184,31 @@ private void reregister(KafkaMbean mbean) {
         }
     }
 
-    private static class KafkaMbean implements DynamicMBean {
+    public int size() {
+        return this.mbeans.size();
+    }
+
+    protected static class KafkaMbean implements DynamicMBean {
+        private final String metricName;
```

Comment:
It allows getting the appropriate `KafkaMBean` from `mbeans` map. But now I think that better approach would have been to update the map itself:
```java
    private final Map<ObjectName, KafkaMbean> mbeans = new HashMap<ObjectName, KafkaMbean>();
```